src/policies/security.ts

```diff
@@ -0,0 +1,233 @@
+ */
+When(a.Pod)
+  .IsCreatedOrUpdated()
+  .Mutate(request => {
+    const pod = request.Raw.spec!;
+
+    // Ensure the securityContext field is defined
+    pod.securityContext = pod.securityContext || {};
+
+    // Set the runAsNonRoot field to true if it is undefined
+    if (pod.securityContext.runAsNonRoot === undefined) {
+      pod.securityContext.runAsNonRoot = true;
+    }
+  })
+  .Validate(request => {
+    // Check if running as root by checking if runAsNonRoot is false or runAsUser is 0
+    const isRoot = (ctx: Partial<V1SecurityContext>) => {
+      const isRunAsRoot = ctx.runAsNonRoot === false;
+      const isRunAsRootUser = ctx.runAsUser === 0;
+
+      return isRunAsRoot || isRunAsRootUser;
+    };
+
+    // Check pod securityContext
+    const podCtx = request.Raw.spec?.securityContext || {};
+    if (isRoot(podCtx)) {
+      return request.Deny("Pod level securityContext does not meet the non-root user requirement.");
+    }
+
+    // Check container securityContext
+    const hasRootContainer = containers(request)
+      .flatMap(c => c.securityContext || {})
+      .find(isRoot);
+
+    if (hasRootContainer) {
+      return request.Deny(
+        `Root user is not allowed. One or more containers in the Pod have a securityContext that does not meet the non-root user requirement.`,
+      );
+    }
+
+    return request.Approve();
+  });
+
+/**
+ * Restrict Proc Mount in Pods
+ *
+ * The default /proc masks are set up to reduce the attack surface. This policy
+ * ensures nothing but the specified procMount can be used. By default only "Default"
+ * is allowed. Applies to all containers, initContainers, and ephemeralContainers in a Pod.
+ *
+ * @related https://repo1.dso.mil/big-bang/product/packages/kyverno-policies/-/blob/main/chart/templates/restrict-proc-mount.yaml
+ */
+When(a.Pod)
+  .IsCreatedOrUpdated()
+  .Validate(request => {
+    const hasOtherProcMount = containers(request)
+      .flatMap(c => c.securityContext || {})
+      .find(ctx => ctx.procMount !== "Default");
+
+    if (hasOtherProcMount) {
+      return request.Deny(`Changing the proc mount from the default is not allowed.`);
+    }
+
+    return request.Approve();
+  });
+
+/**
+ * Restrict Seccomp in Pods
+ *
+ * The SecComp profile should not be explicitly set to Unconfined. This policy ensures
+ * that the `seccompProfile.Type` is undefined or restricted to `RuntimeDefault` or
+ * `Localhost`. Applies to Pods and all types of containers within them.
+ *
+ * @related https://repo1.dso.mil/big-bang/product/packages/kyverno-policies/-/blob/main/chart/templates/restrict-seccomp.yaml
+ */
+When(a.Pod)
+  .IsCreatedOrUpdated()
+  .Validate(request => {
+    const authorizedTyes = ["RuntimeDefault", "Localhost"];
+
+    // Check Pod level security context
+    const podProfileType = request.Raw.spec?.securityContext?.seccompProfile?.type || "";
```

Comment:
I think the `|| ""` is problematic here since the check on the next line requires it to be in the allow list, which `""` is not. Not sure the best fix, adding it to the list or changing this to a default of `RuntimeDefault`?